### DISCUSSION TITLE
Microsoft OneDrive latest version

### DISCUSSION THREAD

**Discussion by Maurice-AxiansNL** [EUCPilots/evergreen-module#105] [Q&A] [answered]
Hi guys,

Small question:

When I run the following:

`$OneDrive = Get-MicrosoftOneDrive | Where-Object { $_.Type -eq "Exe" -and $_.Ring -eq "Production" }` 

The output will be two Exe files, do you have an idea how to download only the highest version?

Thanks

**Comment by msfreaks** [comment, marked as answer]
You could try this:
```powershell
(Get-MicrosoftOneDrive | Where-Object { $_.Type -eq "Exe" -and $_.Ring -eq "Production" }) | Sort-Object -Property @{ Expression = { [System.Version]$_.Version }; Descending = $true } | Select-Object -First 1
```
Will work for however many versions are returned.

**Reply by Maurice-AxiansNL** [reply]
Thanks, this worked like a charm.

**Comment by aaronparker** [comment]
The OneDrive update source doesn't always return multiple Exes, not sure why.

Sorting on the `Version` property will also work without casting it to `[System.Version]`. @msfreaks Have you seen it fail if sorting as is?

```powershell
Get-MicrosoftOneDrive | Where-Object { $_.Type -eq "Exe" -and $_.Ring -eq "Production" } | Sort-Object -Property Version -Descending | Select-Object -First 1
```

**Reply by msfreaks** [reply]
Yes, without casting it will sort as string, for example, 2.1 will be listed as a newer version than 2.9
`@("2.9","2.1") | Sort Descending` lists 2.1 on top. 

Comparing is also definitely an issue if you don't cast.
`"2.9" -ge "2.10"` returns true, `[version]"2.9" -ge [version]"2.10"` returns false..
